SConstruct

```diff
@@ -8,28 +8,30 @@ from lsst.sconsUtils.utils import libraryLoaderEnvironment
 from lsst.utils import getPackageDir
 from lsst.ci.hsc.validate import *
 
-from SCons import Action
-
-class CallAction(Action.FunctionAction):
-    def __init__(self, ident, func, **kw):
-        self.ident = ident
-        self.func = func
-        Action.FunctionAction.__init__(self, self.func, kw)
-
-    # This function returns what is hashed to generate the function signature.
-    # This signature is what scons uses to decide if the build function has
-    # changed, and thus the target should be rebuilt. Unfortunately functools.partial
-    # or lambda cause scons to generate a different signature on each run, and thus
-    # rebuild everything. This function is overloaded here to ensure that a consistent
-    # function signature is generated
-    def get_contents(self, *args, **kwargs):
-        return str(self.ident)
-
-def makeSCons(cls, root, *args, **kwargs):
-    instance = cls(root)
-    return CallAction(str(args) + str(kwargs), functools.partial(instance.scons, *args, **kwargs))
+from SCons.Script import SConscript
+SConscript(os.path.join(".", "bin.src", "SConscript"))
 
 env = Environment(ENV=os.environ)
+env["ENV"]["OMP_NUM_THREADS"] = "1"  # Disable threading; we're parallelising at a higher level
+
+def validate(cls, root, dataId=None, **kwargs):
+    """!Construct a command-line for validation
+
+    @param cls  Validation class to use
+    @param root  Data repo root directory
+    @param dataId  Data identifier dict, or None
+    @param kwargs  Additional key/value pairs to add to dataId
```

Comment:
My analysis package uses a custom class for dataId which inherits from dict but supports
`dataId.copy(foo=1, bar=2)`
It's pretty convenient in this sort of situation.  Maybe we should make a feature request on the butler?